File: creational/typescript/singleton_structured/logger.ts
export class Logger {
  private static _instance: Logger | null = null;
  private logCount: number = 0;
  
  private constructor() {
    console.log('Logger: Initializing logger...');
  }
  
  public static get instance(): Logger {
    if (!Logger._instance) {
      Logger._instance = new Logger();
    }
    return Logger._instance;
  }
  
  public log(message: string): void {
    this.logCount++;
    console.log(`[LOG #${this.logCount}] ${message}`);
  }
  
  public getLogCount(): number { return this.logCount; }
}

File: creational/dart/singleton_structured/logger.dart
class Logger {
  static Logger? _instance;
  
  Logger._() {
    print('Logger: Initializing logger...');
  }
  
  static Logger get instance {
    _instance ??= Logger._();
    return _instance!;
  }
  
  int _logCount = 0;
  
  void log(String message) {
    _logCount++;
    print('[LOG #$_logCount] $message');
  }
  
  int get logCount => _logCount;
}
